cryptoprotocol/src/main/java/org/cryptimeleon/incentivesystem/cryptoprotocol/IncentiveSystem.java

```diff
@@ -0,0 +1,83 @@
+package org.cryptimeleon.incentivesystem.cryptoprotocol;
+
+import org.cryptimeleon.craco.sig.sps.eq.SPSEQSignature;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.IncentivePublicParameters;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.Token;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderKeyPair;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.provider.ProviderPublicKey;
+import org.cryptimeleon.incentivesystem.cryptoprotocol.model.keys.user.UserKeyPair;
+import org.cryptimeleon.math.structures.rings.zn.Zn;
+
+/*
+ * Contains all main algorithms of the T2 incentive system.
```

Comment:
Have to think about it, but mentioning T2 in the code comments tells only a limited number of people something (roughly 6 😂). Internally you could come up with a better referencing/naming scheme. 
* Just calling it _the_ incentive system seems too general 
*

Reply:
I renamed it to inventive system for now, but we should decide for some referencing/naming scheme.